src/libprojectM/ProjectM.hpp

```diff
@@ -27,12 +27,15 @@
 #include "fatal.h"
 
 #ifdef WIN32
+
+#ifdef _MSC_VER
 // libs required for win32
 #pragma comment(lib, "psapi.lib")
 #pragma comment(lib, "kernel32.lib")
 
 #pragma warning(disable : 4244)
```

Comment:
This is the typical "potential data loss" warning when casting to lower precision types. We should simply use `static_cast<>` to silence the warning - or better fix the actual conversions.

Reply:
When trying to see what those warnings exactly were, I had some trouble. Initially, I wasn't getting any. But then I saw that CMake checks if `/W4` can be enabled, which it decides it can't for some reason. Then I configured with `-DCMAKE_CXX_FLAGS="/W4"`, and I got swarmed with warnings, much more than I know what to do about.

I will note, however, that some of these were about not using `/EHsc` despite "using" C++ exceptions (by virtue of using the Microsoft C++ standard library which uses exceptions). When I asked CMake to add `/EHsc` (in `features.cmake`), it decided again that it wouldn't do it and I had to configure with `-DCMAKE_CXX_FLAGS="/EHsc"`.

Reply:
I think the slash in the compiler option causes CMake to bail out as it's not an allowed character in CMake variable names (`enable_cflags_if_supported` uses the flag 1:1 as part of the variable name). I'll fix that by replacing slashes with underscores in the variable name.